lib/jsonapi_compliable/adapters/active_record/base.rb

```diff
@@ -0,0 +1,117 @@
+          scope.page(current_page).per(per_page)
+        end
+
+        # (see Adapters::Abstract#count)
+        def count(scope, attr)
+          if attr.to_sym == :total
+            scope.distinct.count
+          else
+            scope.distinct.count(attr)
+          end
+        end
+
+        # (see Adapters::Abstract#average)
+        def average(scope, attr)
+          scope.average(attr).to_f
+        end
+
+        # (see Adapters::Abstract#sum)
+        def sum(scope, attr)
+          scope.sum(attr)
+        end
+
+        # (see Adapters::Abstract#maximum)
+        def maximum(scope, attr)
+          scope.maximum(attr)
+        end
+
+        # (see Adapters::Abstract#minimum)
+        def minimum(scope, attr)
+          scope.minimum(attr)
+        end
+
+        # (see Adapters::Abstract#resolve)
+        def resolve(scope)
+          scope.to_a
+        end
+
+        # Run this write request within an ActiveRecord transaction
+        # @param [Class] model_class The ActiveRecord class we are saving
+        # @return Result of yield
+        # @see Adapters::Abstract#transaction
+        def transaction(model_class)
+          model_class.transaction do
+            yield
+          end
+        end
+
+        def sideloading_classes
+          {
+            has_many: HasManySideload,
+            has_one: HasOneSideload,
+            belongs_to: BelongsToSideload,
+            many_to_many: ManyToManySideload
+          }
+        end
+
+        def associate(parent, child, association_name, association_type)
+          parent.association(association_name).loaded!
+
+          if [:has_many, :many_to_many].include?(association_type)
+            parent.association(association_name).target |= [child]
+          else
+            parent.association(association_name).target = child
+          end
+        end
+
+        # When a has_and_belongs_to_many relationship, we don't have a foreign
+        # key that can be null'd. Instead, go through the ActiveRecord API.
+        # @see Adapters::Abstract#disassociate
+        def disassociate(parent, child, association_name, association_type)
+          if association_type == :many_to_many
+            parent.send(association_name).delete(child)
+          else
+            # Nothing to do here, happened when we merged foreign key
+          end
+        end
+
+        # (see Adapters::Abstract#create)
+        def create(model_class, create_params)
+          instance = model_class.new(create_params)
+          instance.save
+          instance
+        end
+
+        # (see Adapters::Abstract#update)
+        def update(model_class, update_params)
+          instance = model_class.find(update_params.delete(:id))
+          instance.update_attributes(update_params)
+          instance
+        end
+
+        # (see Adapters::Abstract#destroy)
+        def destroy(model_class, id)
+          instance = model_class.find(id)
+          instance.destroy
```

Comment:
Might want to offer a way to override just this line for a specific model/adapter subclass.  Many times I would want to change this to destroy by setting a `deleted_at` timestamp or something.

Reply:
Completely agree, will be a separate PR though - the premise of splitting out the finder is a bit involved.